lib/types.ts

```diff
@@ -168,10 +178,10 @@ type Selector<TKey extends OnyxKey, TOnyxProps, TReturnType> = (value: OnyxEntry
  * })(Component);
  * ```
  */
-type OnyxEntry<TOnyxValue> = TOnyxValue | null;
+type OnyxEntry<TOnyxValue> = TOnyxValue | null | undefined;
```

Comment:
That is a big change for `withOnyx` and other Onyx methods in terms of typings. Does this type (and `OnyxCollection`) work well in NewDot?

Reply:
@blazejkustra Yeah I know, I tried to do different approaches but at the end they looked worse than this one, it will require some changes in E/App but it would be the one that makes more sense imo. Please have a look at this PR where I apply the changes -> https://github.com/Expensify/App/pull/40367